### ISSUE TITLE
EXCEPTION: Uncaught (in promise): Failed to warm up service null

### ISSUE THREAD

**Issue by BSWG** [driftyco/ionic#10170]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[ x] **2.x**

**I'm submitting a ...**  (check one with "x")
[ x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
When starting the application, this is the console output that I get:
[21:14:50]  console.log: Angular 2 is running in the development mode. Call enableProdMode() to enable the production 
            mode. 
[21:14:50]  console.error: EXCEPTION: Uncaught (in promise): Failed to warm up service 
[21:14:50]  console.error: ORIGINAL STACKTRACE: 
[21:14:50]  console.error: Error: Uncaught (in promise): Failed to warm up service at s 
            (http://10.0.0.5:8100/build/polyfills.js:3:4211) at http://10.0.0.5:8100/build/polyfills.js:3:3961 at 
            Object.callbackFromNative (http://10.0.0.5:8100/cordova.js:295:52) at processMessage 
            (http://10.0.0.5:8100/cordova.js:1119:17) at processMessages (http://10.0.0.5:8100/cordova.js:1142:9) at 
            t.invoke (http://10.0.0.5:8100/build/polyfills.js:3:9094) at Object.onInvoke 
            (http://10.0.0.5:8100/build/main.js:42823:37) at t.invoke (http://10.0.0.5:8100/build/polyfills.js:3:9045) 
            at e.run (http://10.0.0.5:8100/build/polyfills.js:3:6481) at http://10.0.0.5:8100/build/polyfills.js:3:4583 
[21:14:50]  console.error: Unhandled Promise rejection: Failed to warm up service ; Zone: angular ; Task: Promise.then ; 
            Value: Failed to warm up service null 
[21:14:50]  console.warn: Native: deviceready did not fire within 2000ms. This can happen when plugins are in an 
            inconsistent state. Try removing plugins from plugins/ and reinstalling them. 
[21:14:50]  console.log: deviceready has not fired after 5 seconds. 
[21:14:50]  error opening ws message: {"category":"console","type":"log","data":["DEVICE READY FIRED AFTER",3938,"ms"]} 

**Expected behavior:**
No exception at startup.

**Related code:**
I am not sure yet if there is any 'related code' because it just happens at startup.

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

Your system information:

Cordova CLI: 6.5.0 
Ionic Framework Version: 2.0.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.0.0
ios-deploy version: 1.9.0 
ios-sim version: 5.0.8 
OS: macOS Sierra
Node Version: v6.9.2
Xcode version: Xcode 8.2.1 Build version 8C1002

**Comment by jgw96** [comment]
Hey, thanks for using Ionic! To be honest i've never seen this error before, would you mind posting a repo we can use to reproduce this issue?

**Comment by BSWG** [comment]
Hi, actually after doing some more research I found out that it has something to do with Ionic Native's SafariViewController. 

`SafariViewController.isAvailable().then((available: boolean) => {
      if(!available) return;
      SafariViewController.warmUp();
});`

If I remove SafariViewController.warmUp(); the Exception disappears. Under the hood this warmUp() is supposed to warm up Chrome Tabs. But this fails even on an Android device it seems.

Although I resolved the exception for now the warning remains:
[21:14:50] console.warn: Native: deviceready did not fire within 2000ms. This can happen when plugins are in an
inconsistent state. Try removing plugins from plugins/ and reinstalling them.
[21:14:50] console.log: deviceready has not fired after 5 seconds.
[21:14:50] error opening ws message: {"category":"console","type":"log","data":["DEVICE READY FIRED AFTER",3938,"ms"]}
